release.py

```diff
@@ -0,0 +1,694 @@
+
+        logger.info("Let's try again...")
+
+    return new_version
+
+
+def get_new_scenario_version_post_release() -> str:
+    """Get a new post release version for scenario.
+
+    Default value is generated by bumping the current minor version and add the .dev0 suffix.
+
+    Can be overridden with a user-provided version.
+    """
+    current_version, _ = parse_scenario_version()
+    suggested_version = bump_minor_version(current_version) + '.dev0'
+    logger.info(f'Suggested new scenario version: {suggested_version}')
+
+    while True:
+        user_input = input(
+            f'Input post release scenario version (press enter to use suggested version '
+            f"{suggested_version or 'required'}), or 'c' to cancel: "
+        ).strip()
+
+        if user_input.lower() == 'c':
+            logger.warning('Post release canceled.')
+            exit(0)
+
+        suggested_version = user_input if user_input else suggested_version
+
+        if not suggested_version:
+            logger.error('Error: No version specified and no suggestion available')
+            continue
+
+        if not re.match(r'^\d+\.\d+\.\d+\.dev\d+$', suggested_version):
+            logger.error('Error: Version must be in format X.Y.Z.devN')
+            continue
+
+        confirm = (
+            input(
+                f"Confirm using scenario version '{suggested_version}' for post release? [y/N]: "
+            )
+            .strip()
+            .lower()
+        )
+
+        if confirm == 'y':
+            break
+
+        logger.info("Let's try again...")
+
+    return suggested_version
+
+
+def update_versions_for_post_release(repo: github.Repository.Repository, branch_name: str):
+    """Update version files to the post-release version with '.dev0' suffix."""
+    ops_version = get_new_version_post_release(repo, branch_name)
+    scenario_version = get_new_scenario_version_post_release()
+    update_ops_version(ops_version, scenario_version)
+    update_testing_version(ops_version, scenario_version)
+    update_tracing_version(ops_version)
+    update_uv_lock()
+
+
+def check_update_charm_pins_prs(repo: github.Repository.Repository):
+    """Check for open PRs that update charm pins."""
+    prs = repo.get_pulls(state='open')
+    open_prs = [pr for pr in prs if pr.title == 'chore: update charm pins']
+    if open_prs:
+        logger.info('Please merge "update charm pins" PRs first:')
+        pr = open_prs[0]
+        pr_url = f'#{pr.number} - {pr.html_url}'
+        logger.info(f'  {pr_url}')
+        if len(open_prs) > 1:
+            logger.info(
+                f'Note that there are {len(open_prs) - 1} more open update charm pins PRs.'
+            )
+
+
+def draft_release(owner: str, repo_name: str, base_branch: str):
+    """Create a draft release, update changelog, and create a PR for the release."""
+    org = gh_client.get_organization(owner)
+    repo = org.get_repo(repo_name)
+
+    check_update_charm_pins_prs(repo)
```

Comment:
I wonder if we might need an override for this, but I think it's fine to leave as-is for now, and we can add that if it does come up in the future.